### PR TITLE
fix(taskframework): update schedule task status when cancel completed or heart check timeout

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-refactor
module-taskframework
#### What this PR does / why we need it:
1.remove ResultHandleService, replaced by DefaultJobProcessUpdateListener
2.update schedule task status in  `DefaultJobTerminateListener` when cancel completed or heart check timeout
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by guowl3 — DISMISSED** [review]
LGTM

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java [hunk 1]
```diff
@@ -54,5 +59,25 @@ public void onEvent(JobTerminateEvent event) {
                 }
             });
         }
+
+        Optional<ScheduleTaskEntity> scheduleTask = taskRepository.findByJobId(jobEntity.getId());
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java:63]: add a unique key for this column

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java [hunk 2]
```diff
@@ -54,5 +59,25 @@ public void onEvent(JobTerminateEvent event) {
...
+        if (scheduleTask.isPresent() && !scheduleTask.get().getStatus().isTerminated()) {
+            int row =
+                    taskRepository.updateStatusById(scheduleTask.get().getId(), event.getStatus().convertTaskStatus());
+            if (row >= 1) {
+                log.info("Update scheduleTask successfully, scheduleTaskId={}, status={}.", jobEntity.getId(),
+                        event.getStatus().convertTaskStatus());
+            }
+        } else {
+            TaskRepository repository = SpringContextUtil.getBean(TaskRepository.class);
+            Optional<TaskEntity> taskEntity = repository.findById(jobEntity.getId());
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java:73]: findbyJobId~

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java]: add `public Optional<ScheduleTaskEntity> findByJobId(Long jobId) ` for check unique key

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/listener/DefaultJobTerminateListener.java]: done

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**